components/EmotionDetails.js

```diff
@@ -4,7 +4,12 @@ import styled from "styled-components";
 import Link from "next/link";
 import { useState } from "react";
 
-export default function EmotionDetails({ emotion, onDeleteEmotion }) {
+export default function EmotionDetails({
+  emotion,
+  onDeleteEmotion,
+  myBookmarkedEmotions,
+  onToggleBookmark,
+}) {
   const [isDialogOpen, setIsDialogOpen] = useState(false);
```

Comment:
You could consider looking into the [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog) element.